digital-signage-app-v2/app/services/metadata.js

```diff
@@ -0,0 +1,62 @@
+import Ember from 'ember';
+
+export default Ember.Service.extend({
+  _nodes: null,
+  _edges: null,
+  
+  init() {
+    this._super(...arguments);
+    
+    this.clearData();
+  },
+  addNode(node = { }) {
+    this.get('_nodes').pushObject({
+      id: node.id ? node.id : null,
+      prettyName: node.prettyName ? node.prettyName : null,
+      length: node.length ? node.length : null,
+      timeWatched: node.timeWatched ? node.timeWatched : null,
+      attributes: Ember.isArray(node.attributes) ? node.attributes : [ ]
+    });
+  },
+  editNode(id, newData = { }) { // TODO Make not loop
+    this.get('_nodes').forEach((node, index) => {
```

Comment:
This only works assuming a video was watched only once or it is being logged only once. Might want to patch a node at a position rather than by id